src/lib/dnssd/minimal_mdns/RecordData.cpp

```diff
@@ -36,25 +37,30 @@ bool ParseTxtRecord(const BytesRange & data, TxtRecordDelegate * callback)
         {
             return false;
         }
-        pos++;
 
         // name=value string of size length
-        const uint8_t * equalPos = pos;
+        const uint8_t * equalPos = pos + 1;
         while ((*equalPos != '=') && ((equalPos - pos) < length))
```

Comment:
I guess an other way would be to rewrite the while loop as: `while (((equalPos - pos) < length) && (*equalPos != '='))`

Reply:
I think that would be more self-evidently safe....  Let's do that, please.